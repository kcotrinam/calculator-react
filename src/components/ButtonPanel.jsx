import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';
import '../scss/components/panelButtons.scss';

const BUTTONGROUP = {
  group1: ['AC', '+/-', '%', '/'],
  group2: ['7', '8', '9', 'x'],
  group3: ['4', '5', '6', '-'],
  group4: ['1', '2', '3', '+'],
  group5: ['0', '.', '=', '+'],
};

const ButtonPanel = ({ clickHandler }) => {
  const handleClick1 = name => {
    clickHandler(name);
  };

  const renderButtons = group => group.map(name => (
    <Button
      key={name}
      name={name}
      handlerClick={handleClick1}
    />
  ));

  const {
    group1,
    group2,
    group3,
    group4,
    group5,
  } = BUTTONGROUP;

  return (
    <>
      <div className="panel-button">
        <div className="row">{renderButtons(group1)}</div>
        <div className="row">{renderButtons(group2)}</div>
        <div className="row">{renderButtons(group3)}</div>
        <div className="row">{renderButtons(group4)}</div>
        <div className="row">{renderButtons(group5)}</div>
      </div>
    </>
  );
};

ButtonPanel.propTypes = { clickHandler: PropTypes.func.isRequired };
export default ButtonPanel;
