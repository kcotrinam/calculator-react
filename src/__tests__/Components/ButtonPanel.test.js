import { create } from 'react-test-renderer';
import '@testing-library/jest-dom';
import ButtonPanel from '../../components/ButtonPanel';

test('renders correctly', () => {
  const tree = create(<ButtonPanel clickHandler={() => 'test'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
