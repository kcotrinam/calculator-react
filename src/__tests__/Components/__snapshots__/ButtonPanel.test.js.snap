// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="panel-button"
>
  <div
    className="row"
  >
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="btn btn-number"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="btn btn-number"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="btn btn-number"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="btn btn-number"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="btn btn-number"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="btn btn-number"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="btn btn-number"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="btn btn-number"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="btn btn-number"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="btn btn-number btn-zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn btn-equal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
